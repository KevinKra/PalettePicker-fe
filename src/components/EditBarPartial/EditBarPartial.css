.PhraseBlock {
  position: absolute;
  z-index: 2;
  text-align: center;
  width: 100%;
  transition: 0.35s cubic-bezier(0.79, 0.16, 0.28, 1.24);
  font-family: "Lobster", cursive;
  letter-spacing: 1px;
}

.phrase-background {
  margin: auto;
  width: 30%;
  min-width: 200px;
  transform: skew(35deg);
  background-color: coral;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.75);
}

.phrase-background > * {
  transform: skew(-35deg);
}

.phrase-background > h2 {
  margin: 0;
  color: white;
  padding: 0.3rem;
}

.phrase-block-content {
  width: 25%;
  margin: 0.2rem auto;
  min-width: 160px;
}

.phrase-button {
  border: 1px solid white;
  cursor: pointer;
  border-radius: 4px;
  width: 100%;
  background-color: coral;
  color: white;
  font-size: 14px;
}

.current-format {
  display: flex;
  font-family: "Comfortaa", cursive;
  width: 90%;
  margin: 0 auto;
  justify-content: space-around;
  padding: 0.2rem;
  background-color: rgb(40, 48, 56);
}

.current-format > * {
  color: white;
  margin: 0;
  font-size: 12px;
  min-width: 33%;
}

.format-hue {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.format-hue > p {
  margin: 1px 0 0 0;
  min-width: 25px;
  margin-left: 15%;
}

.format-hue > i {
  margin-right: 15%;
}

.format-hue > i:hover {
  cursor: pointer;
}

.pb-active {
  transform: translateY(0%);
}

.pb-inactive {
  transform: translateY(-100%);
}

.locked-btn {
  cursor: auto !important;
  background-color: lightgray;
}
