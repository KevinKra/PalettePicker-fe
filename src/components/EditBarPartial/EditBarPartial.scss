@import "../../styles/main.scss";

.EditBarPartial {
	position: absolute;
	z-index: 2;
	text-align: center;
	width: 100%;
	transition: 0.35s ease-in-out;
	font-family: "Lobster", cursive;
	letter-spacing: 1px;

	.phrase-background {
		margin: auto;
		width: 30%;
		min-width: 200px;
		transform: skew(35deg);
		background-color: coral;
		box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.75);
	}

	.phrase-background > * {
		transform: skew(-35deg);
	}

	.phrase-background > h2 {
		margin: 0;
		color: white;
		padding: 0.3rem;
	}

	.phrase-block-content {
		width: 25%;
		margin: 0.2rem auto;
		min-width: 160px;
	}

	.phrase-button {
		@include btn-style-1;
		text-transform: capitalize;
		font-size: 12px;
		border: 1px solid white;
		width: 100%;
	}

	.current-format {
		display: flex;
		font-family: "Comfortaa", cursive;
		width: 90%;
		margin: 0 auto;
		justify-content: space-around;
		padding: 0.2rem;
		background-color: rgb(40, 48, 56);
	}

	.current-format > * {
		color: white;
		margin: 0;
		font-size: 12px;
		min-width: 33%;
	}

	.format-hue {
		display: flex;
		justify-content: space-evenly;
		align-items: center;
	}

	.format-hue > p {
		margin: 1px 0 0 0;
		min-width: 25px;
		margin-left: 15%;
	}

	.format-hue > i {
		margin-right: 15%;
	}

	.format-hue > i:hover {
		cursor: pointer;
	}

	.pb-active {
		transform: translateY(0%);
	}

	.pb-inactive {
		transform: translateY(-100%);
	}

	.locked-btn {
		cursor: auto;
		background-color: lightgray;
		pointer-events: none;
		&:focus {
			outline: none;
		}
	}
}
