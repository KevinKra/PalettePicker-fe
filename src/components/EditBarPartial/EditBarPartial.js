import React from 'react';
import './EditBarPartial.scss';

export default function EditBarPartial(props) {
	return (
		<section className={`${!props.showFullEditBar ? 'PhraseBlock pb-active' : 'PhraseBlock pb-inactive'}`}>
			<header className="phrase-background">
				<h2>Build Your Palette!</h2>
			</header>
			<section className="phrase-block-content">
				<button
					className={props.hueLocked ? 'phrase-button locked-btn' : 'phrase-button'}
					onClick={(e) => props.updateColors(e, props.colors)}
				>
					Refresh Colors
				</button>
				<div className="current-format">
					<div className="format-hue">
						<p>{props.hue}</p>
						{props.hueLocked ? (
							<i className="fas fa-sm fa-lock" onClick={props.toggleHueLock} />
						) : (
							<i className="fas fa-sm fa-unlock-alt" onClick={props.toggleHueLock} />
						)}
					</div>
					<p>{props.colorScheme}</p>
					<p>{props.variation}</p>
				</div>
			</section>
		</section>
	);
}
