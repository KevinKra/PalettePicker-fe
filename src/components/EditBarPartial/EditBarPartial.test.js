import React from "react";
import EditBarPartial from "./EditBarPartial";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("EditBarPartial", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EditBarPartial hueLocked={true} />);
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should have pb-inactive rendering if props.editable is true", () => {
    wrapper = shallow(<EditBarPartial editable={true} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should have pb-active rendering if props.editable is false", () => {
    wrapper = shallow(<EditBarPartial editable={false} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should lock the 'Refresh Colors' button if hue selection is locked", () => {
    const button = wrapper
      .find("button")
      .at(0)
      .hasClass("locked-btn");
    expect(button).toBe(true);
  });
  it("should unlock the 'Refresh Colors' button if hue selection is unlocked", () => {
    const wrapper = shallow(<EditBarPartial hueLocked={false} />);
    const button = wrapper
      .find("button")
      .at(0)
      .hasClass("locked-btn");
    expect(button).toBe(false);
  });
  it("should display the locked icon if the hueLocked prop is true", () => {
    const icon = wrapper
      .find(".format-hue")
      .children("i")
      .hasClass("fa-lock");
    expect(icon).toEqual(true);
  });
  it("should display the unlocked icon if the hueLocked prop is false", () => {
    const wrapper = shallow(<EditBarPartial hueLocked={false} />);
    const icon = wrapper
      .find(".format-hue")
      .children("i")
      .hasClass("fa-unlock-alt");
    expect(icon).toEqual(true);
  });
});
