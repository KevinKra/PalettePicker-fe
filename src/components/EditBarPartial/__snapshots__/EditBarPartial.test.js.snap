// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBarPartial should have pb-active rendering if props.editable is false 1`] = `
<section
  className="PhraseBlock pb-active"
>
  <header
    className="phrase-background"
  >
    <h2>
      Choose a color!
    </h2>
  </header>
  <section
    className="phrase-block-content"
  >
    <button
      className="phrase-button"
      onClick={[Function]}
    >
      Refresh Colors
    </button>
    <div
      className="current-format"
    >
      <div
        className="format-hue"
      >
        <p />
        <i
          className="fas fa-sm fa-unlock-alt"
        />
      </div>
      <p />
      <p />
    </div>
  </section>
</section>
`;

exports[`EditBarPartial should have pb-inactive rendering if props.editable is true 1`] = `
<section
  className="PhraseBlock pb-inactive"
>
  <header
    className="phrase-background"
  >
    <h2>
      Choose a color!
    </h2>
  </header>
  <section
    className="phrase-block-content"
  >
    <button
      className="phrase-button"
      onClick={[Function]}
    >
      Refresh Colors
    </button>
    <div
      className="current-format"
    >
      <div
        className="format-hue"
      >
        <p />
        <i
          className="fas fa-sm fa-unlock-alt"
        />
      </div>
      <p />
      <p />
    </div>
  </section>
</section>
`;

exports[`EditBarPartial should match the snapshot 1`] = `
<section
  className="PhraseBlock pb-active"
>
  <header
    className="phrase-background"
  >
    <h2>
      Choose a color!
    </h2>
  </header>
  <section
    className="phrase-block-content"
  >
    <button
      className="phrase-button locked-btn"
      onClick={[Function]}
    >
      Refresh Colors
    </button>
    <div
      className="current-format"
    >
      <div
        className="format-hue"
      >
        <p />
        <i
          className="fas fa-sm fa-lock"
        />
      </div>
      <p />
      <p />
    </div>
  </section>
</section>
`;
